presto-verifier/src/main/java/com/facebook/presto/verifier/framework/QueryConfiguration.java

```diff
@@ -57,10 +59,18 @@ public QueryConfiguration applyOverrides(QueryConfigurationOverrides overrides)
         if (overrides.getSessionPropertiesOverrideStrategy() == OVERRIDE) {
             sessionProperties = overrides.getSessionPropertiesOverride();
         }
-        else if (overrides.getSessionPropertiesOverrideStrategy() == SUBSTITUTE) {
-            sessionProperties = new HashMap<>(sessionProperties);
-            for (Entry<String, String> entry : overrides.getSessionPropertiesOverride().entrySet()) {
-                sessionProperties.put(entry.getKey(), entry.getValue());
+        else {
+            if (overrides.getSessionPropertiesOverrideStrategy() == SUBSTITUTE || overrides.getSessionPropertiesOverrideStrategy() == SUBSTITUTE_AND_REMOVE) {
+                sessionProperties = new HashMap<>(sessionProperties);
+                for (Entry<String, String> entry : overrides.getSessionPropertiesOverride().entrySet()) {
+                    sessionProperties.put(entry.getKey(), entry.getValue());
+                }
+            }
+            if (overrides.getSessionPropertiesOverrideStrategy() == REMOVE || overrides.getSessionPropertiesOverrideStrategy() == SUBSTITUTE_AND_REMOVE) {
+                sessionProperties = new HashMap<>(sessionProperties);
+                for (String sessionPropertyToRemove : overrides.getSessionPropertiesRemoval()) {
+                    sessionProperties.remove(sessionPropertyToRemove);
+                }
             }
```

Comment:
I don't think that we need 2 separate settings for this - `SUBSTITUTE_AND_REMOVE` and `REMOVE`. If someone sets something to be removed - lets just remove them irrespective of the approach.

Reply:
Removed the settings.